### PR TITLE
[Enhancement] Add BE metrics to information_schema

### PR DESCRIPTION
**Pull request by decster**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Problem Summary(Required) ：

This PR adds BE's metrics to information_schema

```
mysql> select * from information_schema.be_metrics;
+-------+-----------------------------------------------+-----------------------------------------------------------+--------------+
| BE_ID | NAME                                          | LABELS                                                    | VALUE        |
+-------+-----------------------------------------------+-----------------------------------------------------------+--------------+
| 10003 | active_scan_context_count                     |                                                           |            0 |
| 10003 | binary_column_pool_bytes                      |                                                           |            0 |
| 10003 | bitmap_index_mem_bytes                        |                                                           |            0 |
| 10003 | blocks_created_total                          |                                                           |            0 |
| 10003 | blocks_deleted_total                          |                                                           |            0 |
| 10003 | blocks_open_reading                           |                                                           |            0 |
| 10003 | blocks_open_writing                           |                                                           |            0 |
| 10003 | bloom_filter_index_mem_bytes                  |                                                           |            0 |
| 10003 | broker_count                                  |                                                           |            0 |
| 10003 | brpc_endpoint_stub_count                      |                                                           |            0 |
| 10003 | bytes_read_total                              |                                                           |            0 |
| 10003 | bytes_written_total                           |                                                           |            0 |
| 10003 | central_cache_free_bytes                      |                                                           |       311856 |
...


mysql> select * from information_schema.be_metrics where like(labels, '%mode=user%');
+-------+------+-----------+----------+
| BE_ID | NAME | LABELS    | VALUE    |
+-------+------+-----------+----------+
| 10003 | cpu  | mode=user | 30653342 |
+-------+------+-----------+----------+
1 row in set (0.03 sec)

```
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/catalog/SchemaTable.java [hunk 1]
```diff
@@ -523,6 +523,16 @@ public static Builder builder() {
                                     .column("STATE", ScalarType.createVarchar(NAME_CHAR_LEN))
                                     .column("TYPE", ScalarType.createVarchar(NAME_CHAR_LEN))
                                     .build()))
+                    .put("be_metrics", new SchemaTable(
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/catalog/SchemaTable.java:526]: How can you fetch metrics from all backends? Seem this only chooses one of all backends.

File: be/src/exec/schema_scanner/schema_be_metrics_scanner.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    std::vector<MetricsInfo>& _infos;
+};
+
+Status SchemaBeMetricsScanner::start(RuntimeState* state) {
+    auto o_id = get_backend_id();
+    _be_id = o_id.has_value() ? o_id.value() : -1;
+    _infos.clear();
+    SchemaCoreMetricsVisitor visitor(_infos);
+    StarRocksMetrics::instance()->metrics()->collect(&visitor);
+    LOG(INFO) << strings::Substitute("SchemaBeMetricsScanner #metrics:$0", _infos.size());
```
Comment (imay) [inline be/src/exec/schema_scanner/schema_be_metrics_scanner.cpp:76]: How about removing this log?

File: be/src/exec/schema_scanner/schema_be_metrics_scanner.cpp [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        {"BE_ID", TYPE_BIGINT, sizeof(int64_t), false},
+        {"NAME", TYPE_VARCHAR, sizeof(StringValue), false},
+        {"LABELS", TYPE_VARCHAR, sizeof(StringValue), false},
+        {"VALUE", TYPE_BIGINT, sizeof(int64_t), false},
+};
+
+SchemaBeMetricsScanner::SchemaBeMetricsScanner()
+        : SchemaScanner(_s_columns, sizeof(_s_columns) / sizeof(SchemaScanner::ColumnDesc)) {}
+
+SchemaBeMetricsScanner::~SchemaBeMetricsScanner() = default;
```
Comment (chaoyli) [inline be/src/exec/schema_scanner/schema_be_metrics_scanner.cpp:37]: 1. How long does it collect the metrics from BE?
2. If the metrics need to be persisted, after the FE restarted?

File: fe/fe-core/src/main/java/com/starrocks/catalog/SchemaTable.java (hunk 1 above)
Comment (decster) [reply fe/fe-core/src/main/java/com/starrocks/catalog/SchemaTable.java:526]: if the query doesn't have a filter like `be_id = xx`, the planer will assign a scanrange for each alive BE. If the query does have a fliter like `be_id = x`, the planner will assign a single scan range at that BE.

File: be/src/exec/schema_scanner/schema_be_metrics_scanner.cpp (hunk 2 above)
Comment (decster) [reply be/src/exec/schema_scanner/schema_be_metrics_scanner.cpp:76]: ok

File: be/src/exec/schema_scanner/schema_be_metrics_scanner.cpp (hunk 3 above)
Comment (decster) [reply be/src/exec/schema_scanner/schema_be_metrics_scanner.cpp:37]: >  How long does it collect the metrics from BE?
pretty fast in my test, like in the example.
`1 row in set (0.03 sec)`

> If the metrics need to be persisted, after the FE restarted?
Metrics don't persist currently.

File: gensrc/thrift/Descriptors.thrift [hunk 4]
```diff
@@ -138,6 +138,7 @@ enum TSchemaTableType {
     SCH_TASK_RUNS,
     SCH_VERBOSE_SESSION_VARIABLES,
     SCH_BE_TABLETS,
+    SCH_BE_METRICS,
     SCH_INVALID
```
[the remark this answers is not in the document]
Comment (decster) [reply gensrc/thrift/Descriptors.thrift:142]: ok

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 8 (87.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|7|8|87.50%|[624]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 8 (87.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|7|8|87.50%|[624]|

**Review by imay — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 13 (92.31%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|8|9|88.89%|[628]|
|:large_blue_circle:|com/starrocks/planner/SchemaScanNode.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**